### PR TITLE
fix: Update delegation flow to incorporate new changes

### PR DESCRIPTION
**Pull request by harshal015**
### Description
- Includes API changes around delegation flow.
- No existing UI is affected
- Can be merged only after new BE changes

## Clickup
https://app.clickup.com/t/86cw0z0w1

### CODE REVIEW

**Comment by harshal015** [comment]
the PR can be merged only after new BE changes

File: src/app/core/models/platform/delegation-type.enum.ts [hunk 1]
```diff
@@ -0,0 +1,4 @@
+export enum DelegationType {
+  SHORT_TERM = 'SHORT_TERM',
```
Comment (bistaastha) [inline src/app/core/models/platform/delegation-type.enum.ts:2]: is the string specified on RHS really needed here

**Review by bistaastha — APPROVED** [review]
Minor comment

File: src/app/core/models/platform/delegation-type.enum.ts (hunk 1 above)
Comment (suyashpatil78) [reply src/app/core/models/platform/delegation-type.enum.ts:2]: +1

File: src/app/core/models/platform/platform-employee.model.ts [hunk 2]
```diff
@@ -9,4 +10,5 @@ export interface PlatformEmployee {
   department?: Department;
   department_id?: string;
   org_id: string;
+  delegatees?: Delegatee[];
```
Comment (suyashpatil78) [inline src/app/core/models/platform/platform-employee.model.ts:13]: Why is the key made optional here?

File: src/app/fyle/delegated-accounts/delegated-accounts.page.ts [hunk 3]
```diff
@@ -84,14 +85,11 @@ export class DelegatedAccountsPage {
...
-                  Object.values(delegatee.us).some(
-                    (delegateeProp: User) =>
-                      delegateeProp &&
-                      delegateeProp.toString() &&
-                      delegateeProp.toString().toLowerCase().includes(searchText.toLowerCase())
+              map(({ delegatedAcc }) =>
+                delegatedAcc.filter((delegator: Delegator) =>
+                  Object.values(delegator).some(
+                    (delegatorProp: string) =>
+                      delegatorProp && delegatorProp.toString().toLowerCase().includes(searchText.toLowerCase())
```
Comment (suyashpatil78) [inline src/app/fyle/delegated-accounts/delegated-accounts.page.ts:92]: delegatorProp?.toString().toLowerCase().includes(searchText.toLowerCase()

File: src/app/fyle/notifications/notifications.page.ts [hunk 4]
```diff
@@ -282,7 +281,10 @@ export class NotificationsPage implements OnInit {
       this.notificationForm.controls.notifyOption.setValue(notifyOption);
     });
 
-    this.isDelegateePresent$ = from(this.authService.getEou()).pipe(map((eou) => eou.ou.delegatee_id !== null));
+    this.isDelegateePresent$ = from(this.authService.getEou()).pipe(
+      switchMap((res) => from(this.employeesService.getByParams({ user_id: `eq.${res.us.id}` }))),
```
Comment (suyashpatil78) [inline src/app/fyle/notifications/notifications.page.ts:285]: why is `from` required here?

**Review by suyashpatil78 — APPROVED** [review]

**Review by Chethan-Fyle — APPROVED** [review]

File: src/app/core/models/platform/platform-employee.model.ts (hunk 2 above)
Comment (harshal015) [reply src/app/core/models/platform/platform-employee.model.ts:13]: not needed, will update this

File: src/app/fyle/notifications/notifications.page.ts (hunk 4 above)
Comment (harshal015) [reply src/app/fyle/notifications/notifications.page.ts:285]: `authService.getEou()` returns a promise, converting it to an observable with `from`

Comment (suyashpatil78) [reply src/app/fyle/notifications/notifications.page.ts:285]: @harshal015 `employeesService.getByParams` doesn't require `from`

Comment (suyashpatil78) [reply src/app/fyle/notifications/notifications.page.ts:285]: As it returns observable itself

Comment (harshal015) [reply src/app/fyle/notifications/notifications.page.ts:285]: ohh lol, will update

**Review by mvaishnavi — APPROVED** [review]

**Pull request merged**